apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ClientPool.cs

```diff
@@ -22,39 +22,46 @@
 using Google.Apis.Auth.OAuth2;
 using Google.Cloud.Spanner.V1;
 using Google.Cloud.Spanner.V1.Logging;
-using Grpc.Auth;
-using Grpc.Core;
 
 namespace Google.Cloud.Spanner.Data
 {
 
-    internal static class ClientPool
+    internal class ClientPool
     {
-        private static readonly ConcurrentDictionary<ClientCredentialKey, CredentialClientPool> s_clientPoolByCredential =
+        public static ClientPool Default { get; } = new ClientPool();
```

Comment:
Since there is a shared common one, I assume we want this to be immutable? If so, add `readonly` to `_clientFactory`.